packages/commonwealth/client/scripts/views/modals/session_signin_modal.tsx

```diff
@@ -57,26 +69,43 @@ export const SessionSigninModal = (props: SessionSigninModalProps) => {
             setSelectedWallet={(wallet) => {
               /* do nothing */
             }}
-            accountVerifiedCallback={() => onModalClose()}
+            accountVerifiedCallback={(account, newlyCreated, linked) => {
+              onVerified(account, newlyCreated, linked)
+            }}
             linking={false}
             hasNoWalletsLink={false}
             showResetWalletConnect={wcEnabled}
           />
         </div>
       </div>
     </div>
+      }
+      onClose={onClose}
+      open={true} />
   );
 };
 
-/* TODO: uncomment when shipping
-export const sessionSigninModal = () => {
-  return new Promise<void>((resolve, reject) => {
-    app.modals.create({
-      modal: SessionSigninModal,
-      data: {},
-      completeCallback: () => resolve(),
-      exitCallback: () => reject(new Error('Invalid signature')),
-    });
+export const showSessionSigninModal = () => {
+  const id = uuidv4();
+  const target = document.createElement('div');
+  let root: Root = null;
+
+  target.id = id;
+
+  root = createRoot(target);
```

Comment:
Can we use cw_modal here instead of custom modal code?

Reply:
Good suggestion.